shelley/chain-and-ledger/executable-spec/src/Shelley/Spec/Ledger/TxData.hs

```diff
@@ -403,40 +452,46 @@ pattern TxIn addr index <-
 
 {-# COMPLETE TxIn #-}
 
-instance NoUnexpectedThunks (TxIn crypto)
+deriving via UseIsNormalFormNamed "TxIn" (TxIn crypto v) instance NoUnexpectedThunks (TxIn crypto v)
+
+-- | Parametrized tx output
+-- TODO make v compact too
+data TxOut crypto v where
+  TxOutCompact ::
+    CV crypto v =>
+    {-# UNPACK #-} !BSS.ShortByteString ->
+    v ->
+    TxOut crypto v
```

Comment:
it seems we can no longer unpack the Word64 from the coin if we want this to be parametric, so the UTxO memory size will be a bit bigger (one word per TxOut?). we can maybe make an issue to measure the difference.